sklearnex/covariance/incremental_covariance.py

```diff
@@ -14,76 +14,157 @@
+
     Attributes
     ----------
     location_ : ndarray of shape (n_features,)
         Estimated location, i.e. the estimated mean.
 
     covariance_ : ndarray of shape (n_features, n_features)
         Estimated covariance matrix
+
+    n_samples_seen_ : int
+        The number of samples processed by the estimator. Will be reset on
+        new calls to fit, but increments across ``partial_fit`` calls.
+
+    batch_size_ : int
+        Inferred batch size from ``batch_size``.
+
+    n_features_in_ : int
+        Number of features seen during :term:`fit` `partial_fit`.
     """
 
     _onedal_incremental_covariance = staticmethod(onedal_IncrementalEmpiricalCovariance)
 
-    def __init__(self, batch_size=None):
-        self._need_to_finalize = False  # If True then finalize compute should
-        #      be called to obtain covariance_ or location_ from partial compute data
+    if sklearn_check_version("1.2"):
+        _parameter_constraints: dict = {
+            "store_precision": ["boolean"],
+            "assume_centered": ["boolean"],
+            "batch_size": [Interval(numbers.Integral, 1, None, closed="left"), None],
+            "copy": ["boolean"],
+        }
+
+    get_precision = sklearn_EmpiricalCovariance.get_precision
+    error_norm = wrap_output_data(sklearn_EmpiricalCovariance.error_norm)
+    score = wrap_output_data(sklearn_EmpiricalCovariance.score)
+
+    def __init__(
+        self, *, store_precision=False, assume_centered=False, batch_size=None, copy=True
+    ):
+        self.assume_centered = assume_centered
+        self.store_precision = store_precision
         self.batch_size = batch_size
+        self.copy = copy
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearn.covariance.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
 
     def _onedal_finalize_fit(self):
         assert hasattr(self, "_onedal_estimator")
         self._onedal_estimator.finalize_fit()
         self._need_to_finalize = False
 
+        if self.assume_centered:
+            location = self._onedal_estimator.location_
+            self._onedal_estimator.covariance_ += np.dot(location.T, location)
+            self._onedal_estimator.location_ = np.zeros(
+                location.shape, dtype=location.dtype
+            )
+        if self.store_precision:
+            self.precision_ = linalg.pinvh(
+                self._onedal_estimator.covariance_, check_finite=False
+            )
+        else:
+            self.precision_ = None
+
     def _onedal_partial_fit(self, X, queue):
         onedal_params = {
             "method": "dense",
             "bias": True,
         }
         if not hasattr(self, "_onedal_estimator"):
             self._onedal_estimator = self._onedal_incremental_covariance(**onedal_params)
-        self._onedal_estimator.partial_fit(X, queue)
-        self._need_to_finalize = True
+        try:
```

Comment:
what is the purpose of this change?

Reply:
Though this may come down to your understanding of the onedal side, maybe this is a bad assumption? let me know

Reply:
but if it fails on partial_fit stage then it doesn't make sense to do finalize after, so, setting _need_to_finalize equal to True after that is quite weird for me

Reply:
it makes more sense but i would say that the best way in this case just to let it fail as is and let user decide what to do with it depending on the failure.